### PR TITLE
Gives random event flockminds a free trace as a treat

### PR DESCRIPTION
**Pull request by TobleroneSwordfish**
[BALANCE]
## About the PR 
Flockmind spawned from random events when the server has >40 alive players will immediately attempt to spawn a free (0 compute) trace.
Also allows dead antags to spawn as flocktraces since otherwise the flockmind could run out of ghosts very fast by gaining and losing traces.

## Why's this needed? 
Flock getting destroyed immediately after spawning is a problem, this tries to help with that by giving them more sentience to fight back against early attacks without directly increasing their growth speed or power level too much.

### CODE REVIEW

**Comment by FlameArrow57** [comment]
Kind of against, if Flock is dying upon spawning then I think they just picked a not great spot, plus can get a new trace with converting a computer and making a few drones. Also think traces should be something that need to be built to

Also a possibility the Flockmind isn't wanting any traces

**Comment by TobleroneSwordfish** [comment]
I agree it breaks the rule of traces being directly linked to compute, but what I've seen happening is flock being discovered right around when they get their first trace, then one drone dies and the trace gets booted again, leading to them getting steamrolled shortly afterwards. As far as flockminds not wanting traces goes... Why?

**Comment by FlameArrow57** [comment]
Idk just playstyle, can probably just ignore

**Comment by FlameArrow57** [comment]
Is that issue with spawn choice though, or players happening to find Flock by chance? If it's with finding by choice, does it go back to just increasing Flockmind's early game defense like with sentinels? Dunno how much Flocktraces would help with that

**Comment by TobleroneSwordfish** [comment]
drafting since I found bugs

File: code/mob/living/intangible/flock/flocktrace.dm [hunk 1]
```diff
@@ -85,7 +86,9 @@
 /mob/living/intangible/flock/trace/Life(datum/controller/process/mobs/parent)
 	if (..(parent))
 		return TRUE
-	if (src.flock && src.flock.total_compute() < src.flock.used_compute)
+	var/datum/abilityHolder/flockmind/aH = src.abilityHolder
+	aH?.updateCompute()
```
[the remark this answers is not in the document]
Comment (TobleroneSwordfish) [reply code/mob/living/intangible/flock/flocktrace.dm]: that's what I get for resolving conflicts on github

File: code/procs/antagonist_procs.dm [hunk 2]
```diff
@@ -293,9 +293,17 @@
 	synd_mob.implant.Add(M)
 	M.implanted(synd_mob)
 
+/proc/alive_player_count()
+	for(var/client/C)
```
Comment (ZeWaka) [inline code/procs/antagonist_procs.dm:297]: will return null instead of 0

**Review by ZeWaka — CHANGES_REQUESTED** [review]
x

**Pull request merged**